node/node_test.go

```diff
@@ -597,3 +600,65 @@ func TestAPIGather(t *testing.T) {
 		}
 	}
 }
+
+func TestWebsocketHTTPOnSamePort_WebsocketRequest(t *testing.T) {
+	node := startHTTP(t)
+	defer node.stopHTTP()
+
+	wsReq, err := http.NewRequest(http.MethodGet, "http://127.0.0.1:7453", nil)
+	if err != nil {
+		t.Error("could not issue new http request ", err)
+	}
+	wsReq.Header.Set("Connection", "upgrade")
+	wsReq.Header.Set("Upgrade", "websocket")
+	wsReq.Header.Set("Sec-WebSocket-Version", "13")
+	wsReq.Header.Set("Sec-Websocket-Key", "SGVsbG8sIHdvcmxkIQ==")
+
+	wsResponses := make(chan *http.Response)
+	go doHTTPRequest(wsResponses, wsReq, t)
+	wsResponse := <-wsResponses
```

Comment:
You're not really using the channel for anything here, consider simply calling doHTTPRequest.